client/app/vps/cloud-database/vps-cloud-database.controller.js

```diff
@@ -0,0 +1,148 @@
+class VpsCloudDatabaseCtrl {
+  constructor(
+    $q,
+    $stateParams,
+    $timeout,
+    $translate,
+    $window,
+    CloudMessage,
+    ControllerHelper,
+    OvhApiHostingPrivateDatabase,
+    VpsService,
+  ) {
+    this.$q = $q;
+    this.$stateParams = $stateParams;
+    this.$timeout = $timeout;
+    this.$translate = $translate;
+    this.$window = $window;
+    this.CloudMessage = CloudMessage;
+    this.ControllerHelper = ControllerHelper;
+    this.ApiPrivateDb = OvhApiHostingPrivateDatabase.v6();
+    this.ApiWhitelist = OvhApiHostingPrivateDatabase.Whitelist().v6();
+    this.VpsService = VpsService;
+  }
+
+  $onInit() {
+    this.serviceName = this.$stateParams.serviceName;
+
+    this.statusFilterOptions = {
+      values: _.reduce(
+        ['detached', 'restartPending', 'startPending', 'started', 'stopPending', 'stopped'],
+        (result, key) => _.assign(result, { [key]: this.$translate.instant(`common_database_status_${key}`) }),
+        {},
+      ),
+    };
+
+    this.ipv4 = null;
+    this.cloudDatabases = [];
+
+    this.refresh();
+  }
+
+  refresh() {
+    this.loading = true;
+    return this.loadIps()
+      .then((response) => {
+        this.ipv4 = _(response.results).chain()
```

Comment:
Avoid iterating over `String`s
```js
then({ results } => {
    if (_(results).isString()) {
        throw new Error("Temporary error from the API");
    }

    const searchResult = results.find((result) => result.version === 'v4');
    this.ipv4 = _(searchResult).get('ipAddress');
})
```

Reply:
I don't get it. `response.results` is always an array.

Reply:
Not if the API is down and returns a `String` as an error message :)

It has already happened